tests/platform/mellanox/check_sysfs.py

```diff
@@ -34,46 +34,138 @@ def check_sysfs(dut):
         try:
             fan_max_content = dut.command("cat %s" % fan_max)
             fan_max_speed = int(fan_max_content["stdout"])
             assert fan_max_speed > 10000, "Bad fan maximum speed: %s" % str(fan_max_speed)
         except Exception as e:
-            assert False, "Get content from %s failed, exception: %s" % (fan_max, repr(e))
+            assert "Get content from %s failed, exception: %s" % (fan_max, repr(e))
+
+        fan_speed_set = "/var/run/hw-management/thermal/fan{}_speed_set".format(fan_id)
+        fan_speed_set_content = dut.command("cat %s" % fan_speed_set)
+        assert fan_speed_set_content["stdout"] == "153", "Fan speed should be set to 60%, 153/255"
+        fan_set_speed = int(fan_speed_set_content["stdout"])
 
-    fan_speed_get_list = ["/var/run/hw-management/thermal/fan%d_speed_get" % fan_id for fan_id in range(1, fan_count + 1)]
-    for fan_speed_get in fan_speed_get_list:
+        fan_speed_get = "/var/run/hw-management/thermal/fan{}_speed_get".format(fan_id)
         try:
             fan_speed_get_content = dut.command("cat %s" % fan_speed_get)
             fan_speed = int(fan_speed_get_content["stdout"])
-            assert fan_speed > 1000, "Bad fan speed: %s" % str(fan_speed)
+            
         except Exception as e:
-            assert False, "Get content from %s failed, exception: %s" % (fan_speed_get, repr(e))
+            assert "Get content from %s failed, exception: %s" % (fan_speed_get, repr(e))
+        assert fan_min_speed < fan_speed < fan_max_speed, "Bad fan speed: %s" % str(fan_speed)
+        max_tolerence_speed = ((float(fan_set_speed)/256)*fan_max_speed)*(1 + 0.3)
+        min_tolerence_speed = ((float(fan_set_speed)/256)*fan_max_speed)*(1 - 0.3)
+        assert min_tolerence_speed < fan_speed < max_tolerence_speed, "Speed out of tolerence speed range (%d, %d)" % (min_tolerence_speed, max_tolerence_speed)
+        
+    cpu_pack_count = SWITCH_MODELS[dut_hwsku]["cpu_pack"]["number"]
+    if cpu_pack_count != 0:
+        cpu_pack_temp_file = "/var/run/hw-management/thermal/cpu_pack"
+        cpu_pack_temp_file_output = dut.command("cat %s" % cpu_pack_temp_file)
+        cpu_pack_temp = float(cpu_pack_temp_file_output["stdout"])/1000
+        
+        cpu_pack_max_temp_file = "/var/run/hw-management/thermal/cpu_pack_max"
+        cpu_pack_max_temp_file_output = dut.command("cat %s" % cpu_pack_max_temp_file)
+        cpu_pack_max_temp = float(cpu_pack_max_temp_file_output["stdout"])/1000
+        
+        cpu_pack_crit_temp_file = "/var/run/hw-management/thermal/cpu_pack_crit"
+        cpu_pack_crit_temp_file_output = dut.command("cat %s" % cpu_pack_crit_temp_file)
+        cpu_pack_crit_temp = float(cpu_pack_crit_temp_file_output["stdout"])/1000
+        
+        assert cpu_pack_max_temp <= cpu_pack_crit_temp, "Bad cpu pack max temp or crit temp, %s, %s " % (str(cpu_pack_max_temp), str(cpu_pack_crit_temp))
```

Comment:
replace cpu w/ CPU in log message.
in log message don't use abbreviations. replace crit with critical

Reply:
fixed